### PR TITLE
Add a flag to allow skipping Github release creation

### PR DESCRIPTION
**Pull request by akphi**
Fixes https://github.com/changesets/action/issues/87

Added a flag called `SKIP_CREATE_GITHUB_RELEASE` to allow skipping publishing release to Github

@Andarist If you have other suggestions on the config name or the way I pass down the `boolean` please let me know I can fix it ASAP. Would be really appreciative if you could spare some time to look at this. Thank you so much!

Also, I thought about adding some documentation for this on the `README.md` file, but not sure how to organize that, do you have any suggestion here?

### CODE REVIEW

**Comment by Andarist** [comment]
@akphi could you describe your use case that requires introducing this option?

**Comment by akphi** [comment]
@Andarist My use case is fairly similar to that of [babel[(https://github.com/babel/babel/tree/main/packages): I have a monorepo with many workspaces/packages. If I were to publish them, I would accumulate hundreds of tags and releases on Github. I would like to use the Github release page a little more sparingly. Ideally, I want to create an accumulated release note per release, but that's off the scope of this feature request.

I could temporarily workaround for now, by setting a post-release script to delete created tags and releases, but it's a hack 😅.

**Comment by akphi** [comment]
Btw, @Andarist I saw your change in https://github.com/changesets/changesets/pull/701 to add `--no-git-tag`. It will help at least not creating the tags, but the releases are still being created by `changesets/action` right?

**Comment by Andarist** [comment]
Would [fixed](https://github.com/changesets/changesets/pull/690) packages and [aggregated releases](https://github.com/changesets/changesets/issues/683) address your needs? Note that I'm planning to introduce those aggregated releases only for linked and fixed packages - at least for now. Introducing that for regular packages is quite problematic when it comes to changeset creation, sorting etc.

> Btw, @Andarist I saw your change in changesets/changesets#701 to add --no-git-tag. It will help at least not creating the tags, but the releases are still being created by changesets/action right?

I think that releases should not be created with this flag (but I have not tested it). It's because releases are created based on the created tags:
https://github.com/changesets/action/blob/37c2edd85bf5f2aa8edb44ce4ea00eb4a55aa9d4/src/run.ts#L88-L114

That being said - the flag that you are asking for is still useful for the use case described in #87. The author of that issue would like to skip creating github releases for their prereleases.

Could you rebase this PR on top of the current main branch? In the process please remove the `dist/index.js` file - since it's no longer committed to the repo.

File: src/index.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ const getOptionalInput = (name: string) => core.getInput(name) || undefined;
 
 (async () => {
   let githubToken = process.env.GITHUB_TOKEN;
+  let skipCreateGithubRelease = Boolean(process.env.SKIP_CREATE_GITHUB_RELEASE);
```
Comment (Andarist) [inline src/index.ts:11]: i think that the input option for the action would make more sense and I would call it `createGithubReleases`. It's default value would be `true` so you could disable it with `false`.

File: src/run.ts [hunk 2]
```diff
@@ -106,10 +113,14 @@ export async function runPublish({
...
-          pkg,
-          tagName: `${pkg.packageJson.name}@${pkg.packageJson.version}`,
-        })
+        createRelease(
+          octokit,
+          {
+            pkg,
+            tagName: `${pkg.packageJson.name}@${pkg.packageJson.version}`,
+          },
+          skipCreateGithubRelease
```
Comment (Andarist) [inline src/run.ts:122]: It imho makes more sense to call `createRelease` conditionally, rather than to skip releasing within this function.

Comment (akphi) [reply src/run.ts]: The whole content of the `createRelease` function is about creating the Github release, so I used a `guarded if`, but perhaps, putting the whole thing in an `if` block is a bit more future proof 😅

File: src/index.ts (hunk 1 above)
Comment (akphi) [reply src/index.ts]: That's a fair point, I also figure this is a bit specific for this use case so polluting function param list with this boolean flag is not great, hence I wrapped it in `options`. If you think that's overkill, let me know.

**Comment by akphi** [comment]
What about documentation, I felt like we should mention it somewhere if possible.

**Comment by akphi** [comment]
Due to the change in base branch `master -> main`, I have to recreate the PR - https://github.com/changesets/action/pull/130

**Comment by Andarist** [comment]
> Due to the change in base branch master -> main, I have to recreate the PR - #130

Just a tip - the target branch can usually be changed after clicking Edit. You can change the PR's title there but also the target branch.

**Pull request closed without merging**